src/afl-showmap.c

```diff
@@ -1012,12 +1011,30 @@ int main(int argc, char **argv_orig, char **envp) {
 
   }
 
+
```

Comment:
below here you move when what is happening. why?

Reply:
The `detect_file_args()` call is a very side-effect-heavy function. It both replaces the "@@" symbols in the argv with its second arg (`stdin_file`) and also sets the boolean in its third arg (`fsrv->use_stdin`) if it finds any.

The previous code was sending "" for the second arg to get the third arg side effect without replacing the @@ symbols, and then it would manually replace the symbols itself later (former line 1048). This required `detect_file_args()` to have yet another special option (which I removed) to tolerate a blank "" for arg 2, and it also got the @@ substitution wrong (or at least different than what `detect_file_args()` would have done) by only substituting whole indexes from argv.

I moved the "determine `stdin_file`" code up here so that `detect_file_args()` could be called in the "normal" way that it is used in other afl utils and get both "@@ replacement" and "stdin detection" effects in one pass.